browser/brave_rewards/add_funds_popup.cc

```diff
@@ -0,0 +1,562 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include "brave/browser/brave_rewards/add_funds_popup.h"
+
+#include <string>
+
+#include "base/base64.h"
+#include "base/json/json_writer.h"
+#include "base/values.h"
+#include "brave/browser/ui/views/location_bar/brave_location_bar_view.h"
+#include "brave/components/brave_rewards/browser/rewards_service.h"
+#include "brave/components/brave_shields/common/brave_shield_constants.h"
+#include "brave/components/toolbar/constants.h"
+#include "chrome/browser/content_settings/host_content_settings_map_factory.h"
+#include "chrome/browser/profiles/profile.h"
+#include "chrome/browser/ui/browser.h"
+#include "chrome/browser/ui/browser_finder.h"
+#include "chrome/browser/ui/views/frame/browser_view.h"
+#include "components/content_settings/core/browser/host_content_settings_map.h"
+#include "components/guest_view/browser/guest_view_base.h"
+#include "content/public/browser/page_navigator.h"
+#include "content/public/browser/web_contents.h"
+#include "content/public/browser/web_contents_delegate.h"
+#include "content/public/common/referrer.h"
+#include "net/base/escape.h"
+#include "third_party/blink/public/platform/web_referrer_policy.h"
+#include "ui/base/page_transition_types.h"
+#include "ui/display/display.h"
+#include "ui/display/screen.h"
+#include "ui/views/widget/widget.h"
+
+using content::WebContents;
+
+namespace {
+
+constexpr int kPopupPreferredHeight = 800;
+constexpr int kPopupPreferredWidth = 1100;
+
+// Referrer.
+const char kRewardsHost[] = "rewards";
+
+// URL to open in the popup.
+const char kAddFundsUrl[] = "https://uphold-widget.herokuapp.com/index.php";
```

Comment:
are these `herokuapp` URLs going to be changed to brave.com URLs?

Reply:
I think the current plan is to stick with heroku. @bsclifton and @w0ts0n had a discussion on this.

Reply:
i'm a bit nervous about that since we've had 2 or 3 times in the past where people have de-registered heroku domains, forgot they were still being used somewhere, and had attackers re-register those domains.

Reply:
also there's a phishing concern mentioned here: https://github.com/brave/internal/issues/451#issuecomment-438885434

Reply:
The open request for `uphold-widget.brave.com` can be tracked here https://github.com/brave/devops/issues/406 😄

Reply:
That doesn't help Yan's concern. She is saying she is nervous using heroku because if the app get's accidentally taken down or goes offline, someone can steal the app name and basically own uphold-widget.brave.com. I'm saying, we could add a pingdom check to be aware if that happens.

Reply:
@w0ts0n  ohhhhh- I gotcha... there is no way to make it private? or use a static IP?

Reply:
Not easily. One of the limitations of Heroku. No IP addresses and anyone can register a name ones it's been deleted. Pingdom check will probably be fine. Devops is ultra sensitive to it now.

Reply:
https://uphold-widget.brave.com/ is now set up!

Reply:
Yes, pingdom check is in place too.